src/validation_workflow/yum/validation_yum.py

```diff
@@ -0,0 +1,76 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import time
+
+from system.temporary_directory import TemporaryDirectory
+from validation_workflow.api_test_cases import ApiTestCases
+from validation_workflow.download_utils import DownloadUtils
+from validation_workflow.validation import Validation
+from validation_workflow.validation_args import ValidationArgs
+
+
+class ValidateYum(Validation, DownloadUtils):
+
+    def __init__(self, args: ValidationArgs) -> None:
+        super().__init__(args)
+        self.base_url = "https://artifacts.opensearch.org/releases/bundle/"
+        self.tmp_dir = TemporaryDirectory()
+
+    def download_artifacts(self) -> bool:
+        for project in self.args.projects:
+            url = f"{self.base_url}{project}/{self.args.version[0:1]}.x/{project}-{self.args.version[0:1]}.x.repo"
+            if ValidateYum.is_url_valid(url) and ValidateYum.download(url, self.tmp_dir):
+                logging.info(f"Valid URL - {url} and Download Successful !")
+            else:
+                logging.info(f"Invalid URL - {url}")
+                raise Exception(f"Invalid url - {url}")
+        return True
+
+    def installation(self) -> bool:
+        try:
+            os.chdir(self.tmp_dir.path)
+            os.system('sudo rpm --import https://artifacts.opensearch.org/publickeys/opensearch.pgp')
+            for project in self.args.projects:
+                os.system(f'sudo yum remove {project} -y')
+                logging.info('Removed previous versions of Opensearch')
+                urllink = f"{self.base_url}{project}/{self.args.version[0:1]}.x/{project}-{self.args.version[0:1]}.x.repo -o /etc/yum.repos.d/{project}-{self.args.version[0:1]}.x.repo"
+                os.system(f'sudo curl -SL {urllink}')
+                os.system(f"sudo yum install '{project}-{self.args.version}' -y")
```

Comment:
Same as above `Why not use src/system/execute.py instead of os.system?`

Reply:
Replaced os.system implementations with execute.py Sayali. Thanks.